3/src/components/ApolloExample.vue

```diff
@@ -0,0 +1,193 @@
+<template>
+  <div class="apollo-example">
+    <!-- Cute tiny form -->
+    <div class="form">
+      <label for="field-name" class="label">Name</label>
+      <input
+        v-model="name"
+        placeholder="Type a name"
+        class="input"
+        id="field-name"
+      />
+    </div>
+
+    <!-- Apollo watched Graphql query -->
+    <ApolloQuery
+      :query="require('../graphql/HelloWorld.gql')"
+      :variables="{ name }"
+    >
+      <template slot-scope="{ result: { loading, error, data } }">
+        <!-- Loading -->
+        <div v-if="loading" class="loading apollo">Loading...</div>
+
+        <!-- Error -->
+        <div v-else-if="error" class="error apollo">An error occured</div>
+
+        <!-- Result -->
+        <div v-else-if="data" class="result apollo">{{ data.hello }}</div>
+
+        <!-- No result -->
+        <div v-else class="no-result apollo">No result :(</div>
+      </template>
+    </ApolloQuery>
+
+    <!-- Tchat example -->
+    <ApolloQuery :query="require('../graphql/Messages.gql')">
+      <ApolloSubscribeToMore
+        :document="require('../graphql/MessageAdded.gql')"
+        :update-query="onMessageAdded"
+      />
+
+      <div slot-scope="{ result: { data } }">
+        <template v-if="data">
+          <div
+            v-for="message of data.messages"
+            :key="message.id"
+            class="message"
+          >
+            {{ message.text }}
+          </div>
+        </template>
+      </div>
+    </ApolloQuery>
+
+    <ApolloMutation
+      :mutation="require('../graphql/AddMessage.gql')"
+      :variables="{
+        input: {
+          text: newMessage
+        }
+      }"
+      class="form"
+      @done="newMessage = ''"
+    >
+      <template slot-scope="{ mutate }">
+        <form v-on:submit.prevent="formValid && mutate()">
```

Comment:
```suggestion
        <form @submit.prevent="formValid && mutate()">
```

Reply:
Also consider to move logic into a method:
```suggestion
        <form @submit.prevent="onSubmit">
```
Then somewhere in your methods:
```js
export default {
  // ...
  methods: {
    onSubmit() {
      return this.formValid && this.mutate()
    }
  }
}
```